dagger/utilities/dbt_config_parser.py

```diff
@@ -26,116 +28,58 @@ def __init__(self, default_config_parameters: dict):
             data = f.read()
         self._manifest_data = json.loads(data)
         profile_yaml = yaml.safe_load(open(dbt_profile_path, "r"))
-        prod_dbt_profile = profile_yaml[self._dbt_project_dir.split("/")[-1]]["outputs"]["data"]
+        prod_dbt_profile = profile_yaml[self._dbt_project_dir.split("/")[-1]][
+            "outputs"
+        ][self._dbt_profile]
         self._default_data_dir = prod_dbt_profile.get(
             "s3_data_dir"
         ) or prod_dbt_profile.get("s3_staging_dir")
 
-    def generate_io(self, model_name: str) -> Tuple[List[Dict], List[Dict]]:
+    def _generate_dagger_dependency(self, node: dict) -> List[Dict]:
         """
-        Generates the dagger inputs and outputs for the respective dbt model
+        Generates the dagger task based on whether the DBT model node is a staging model or not.
+        If the DBT model node represents a staging model, then a dagger athena task is generated for each source of the DBT model.
+        If the DBT model node is not a staging model, then a dagger athena task and an s3 task is generated for the DBT model node itself.
         Args:
-            model_name: name of the dbt model
+            node: The extracted node from the manifest.json file
 
         Returns:
-            tuple[List[Dict], List[Dict]]: dagger inputs and outputs for the respective dbt model
+            List[Dict]: The respective dagger tasks for the DBT model node
 
         """
-        model_parents = self._get_dbt_model_parents(model_name)
-        model_dagger_inputs = self.generate_dagger_inputs(model_parents)
-        model_dagger_outputs = self.generate_dagger_outputs(
-            model_parents["model_name"],
-            model_parents["schema"],
-            model_parents["relative_s3_path"],
-        )
+        model_name = node["name"]
 
-        return model_dagger_inputs, model_dagger_outputs
+        s3_task = S3_TASK_BASE.copy()
+        dagger_tasks = []
 
-    def parse_dbt_staging_model(self, dbt_staging_model: str) -> Tuple[str, str]:
-        """
-        Parses the dbt staging model to get the core schema and table name
-        Args:
-            dbt_staging_model: name of the DBT staging model
+        if model_name.startswith("stg_"):
```

Comment:
Is this a Choco specific naming convention or Dbt specific?

Reply:
dbt recommends naming staging models this way
<img width="936" alt="image" src="https://github.com/chocoapp/dataeng-dagger/assets/16002904/5e617fff-139d-45cd-8e2f-ef63f56efcb3">